docs/require-by-string-aliases.md

```diff
@@ -0,0 +1,251 @@
+# Require by String with Aliases
+
+## Summary
+
+We need to add intuitive alias and paths functionality to facilitate the grouping together of related Luau files into libraries and allow for future package managers to be developed and integrated easily. 
+
+## Motivation
+
+The Roblox engine does not currently support require-by-string. One motivation for this RFC is to consolidate require syntax and functionality between the Roblox engine and the broader Luau language itself.
+
+Luau itself currently supports a basic require-by-string syntax that allows for requiring Luau modules by relative or absolute path. Unfortunately, the current implementation does not support aliases.
+
+## Design
+
+#### Aliases
+
+Aliases can be used to bind an absolute or relative path to a convenient, case-sensitive name that can be required directly.
+
+```json
+"aliases": {
+    "Roact": "C:/LuauModules/Roact-v1.4.2"
+}
+```
```

Comment:
hmm. I'm actually not sure about case-sensitivity! Paths on Windows and Mac are not case sensitive, but they are on Linux. What do other languages do?

Reply:
I think we should be case insensitive across all platforms, _including Linux_. If we, or the community, develops a package manager, the name of the package they wish to install are going to be case insensitive, so it makes sense for the require by alias to be insensitive.

Otherwise you run into security issues because you typed in the wrong capitalization.

Reply:
Similarly, we want the name of the library to match the name everywhere except by case, because otherwise you have problems found with `pip`, e.g. `No module named 'yaml'` -> `pip install yaml` -> `cannot find yaml` -> google the name of the package -> `pip install pyyaml`.

Reply:
By name do you mean the key in the alias map? If so that's user chosen.

Reply:
Right, it _can be_ user-chosen, but I imagine that if a package management tool exists, it also needs to modify this alias map, right? There's no other way around it. The point I was trying to get across is, the mismatch of "name of the missing module" and "name of the module you need to install" should be exactly the same, _except_ by case.

Reply:
1. Ok if we see this alias map primarily being modified by a package manager then I agree with 1. 
2. I don't see a good way to resolve this. Ideally the package manager assigns it the proper name, but inferring the alias from the path has its problems. Maybe I am not understanding your suggestion. What does "we want the name of the library to match the name everywhere except by case" mean?

Reply:
For posterity, we have decided to make alias names case insensitive and restrict the character set to letters, numbers, and the characters `._-`